src/main/java/matchmaker/backend/controllers/ChallengeController.java

```diff
@@ -110,18 +112,72 @@ public ResponseEntity<Challenge> createChallenge(@RequestBody Challenge newChall
         return ResponseEntity.status(HttpStatus.OK).body(savedChallenge);
     }
 
+    /**
+     * / Get 10 challenges per page by search criteria (filters)
+     * / The method will return a Page object with the challenges for that page
+     *
+     * @param query   - words to search for in the title, tags, description and summary
+     * @param company - the company name to search for
+     * @param branche - the branche name to search for
+     * @param sort    - the sort order, can be Newest_first or deadline_closest_first
+     *                / if empty, Newest_first will be used
+     * @param page    - the page number to return
+     *                / if empty, the first page (0) will be returned
+     */
     @GetMapping("/challenge/search")
-    public Iterable<Challenge> search(
+    public Page<Challenge> search(
             @RequestParam(value = "query", required = false) String query,
             @RequestParam(value = "company", required = false) List<String> company,
             @RequestParam(value = "branche", required = false) List<String> branche,
-            @RequestParam(value = "sort", defaultValue = "Newest_first") String sort) {
+            @RequestParam(value = "sort", defaultValue = "Newest_first") String sort,
+            @RequestParam(value = "page", defaultValue = "0") int page) {
+        Specification<Challenge> specification = getSpecification(query, company, branche);
+
+        //sort the results in this order: the most recent ones and the ones where the deadline is closest
+        Sort sortOrder = Sort.by(Sort.Direction.ASC, "createdAt");
+
+        if ("Newest_first".equalsIgnoreCase(sort)) {
+            sortOrder = Sort.by(Sort.Direction.DESC, "createdAt");
+        } else if ("deadline_closest_first".equalsIgnoreCase(sort)) {
+            sortOrder = Sort.by(Sort.Direction.ASC, "endDate");
+        }
+
+        int pageSize = 10;
```

Comment:
Is dit een waarde die nooit verandert of zijn we in de toekomst van plan om ook aan te geven hoeveel challenges op een pagina mogen vanuit de frontend?